src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -878,7 +878,30 @@ private static bool IsGlobalAttributeTarget(SyntaxToken token)
 
         private bool IsPossibleAttributeDeclaration()
         {
-            return this.CurrentToken.Kind == SyntaxKind.OpenBracketToken;
+            // Have to at least start with `[` to be an attribute
+            if (this.CurrentToken.Kind != SyntaxKind.OpenBracketToken)
+                return false;
+
+            using var _ = this.GetDisposableResetPoint(resetOnDispose: true);
+
+            // Eat the `[`
+            EatToken();
+
+            // `[ id` is always an attribute.
+            if (this.IsTrueIdentifier())
+                return true;
+
+            // `[ word: ...` could be an attribute.
+            if (IsAttributeTarget())
+                return true;
```

Comment:
Are these two checks necessary? If not, is the thought that they make this code's intent more clear?

Reply:
the ISAttributeTarget check is *definitely* necessary.  As `true` can both be a literal, or an attribute-target (targets are any *word*, including keywords).  

The IsTrueIdentifier is technically *not* necessary.  But i prefer the clarity.  This means that the code says: if this is definitely an attribute, continue parsing as such.  If it is def not an attribute, then parse as a collection in some well known cases.